sophiread/SophireadCLI/src/venus_auto_reducer.cpp

```diff
@@ -0,0 +1,257 @@
+
+    // NOTE: we need to process files sequentially as we are accumulating the counts to the
+    //       same set of tiff files in the output folder
+    for (const auto& entry : fs::directory_iterator(input_dir)) {
+        if (entry.is_regular_file() && entry.path().extension() == ".tpx3") {
+            std::string filename = entry.path().stem().string();
+
+            if (processed_files.find(filename) != processed_files.end()) {
+                spdlog::debug("Skipping already processed file: {}", filename);
+                continue;
+            }
+
+            spdlog::info("Processing file: {}", entry.path().string());
+
+            try {
+                // Read the TPX3 file
+                auto raw_data = sophiread::timedReadDataToCharVec(entry.path().string());
+                
+                // Find TPX3 headers
+                auto batches = sophiread::timedFindTPX3H(raw_data);
+                
+                // Process the data
+                sophiread::timedLocateTimeStamp(batches, raw_data);
+                sophiread::timedProcessing(batches, raw_data, config);
+                
+                // Generate output file name
+                std::string output_file = fs::path(output_dir) / (tiff_base + "_bin_xxxx.tiff");
+                
+                // Create TOF images
+                auto tof_images = sophiread::timedCreateTOFImages(batches, config.getSuperResolution(), config.getTOFBinEdges(), tof_mode);
+                
+                // Save TOF images
+                sophiread::timedSaveTOFImagingToTIFF(output_dir, tof_images, config.getTOFBinEdges(), tiff_base);
+                
+                spdlog::info("Processed and saved: {}", output_file);
+
+                // record processed file
+                processed_files.insert(entry.path().stem().string());
+            } catch (const std::exception& e) {
+                spdlog::error("Error processing file {}: {}", entry.path().string(), e.what());
+            }
+        }
+    }
+
+    // Create a string to hold the formatted output
+    std::ostringstream oss;
+    oss << "Processed files: [";
+
+    // Loop through the set and concatenate elements
+    for (auto it = processed_files.begin(); it != processed_files.end(); ++it) {
+        if (it != processed_files.begin()) {
+            oss << ", "; // Add a comma before each element except the first
+        }
+        oss << *it;
+    }
+    oss << "]";
+    spdlog::info(oss.str());
+}
+
+void monitor_directory(const std::string& input_dir, const std::string& output_dir, 
+                       const std::string& tiff_base, const std::string& tof_mode,
+                       const IConfig& config, std::unordered_set<std::string>& processed_files,
+                       int check_interval) {
+    spdlog::info("Starting directory monitoring: {}", input_dir);
+    spdlog::info("Check interval: {} seconds", check_interval);
+
+    while (true) {
```

Comment:
The "proper" way to do this is to use inotify ([man page](https://www.man7.org/linux/man-pages/man7/inotify.7.html)). However, polling the filesystem with appropriate interval is much simpler.